File: modules/celebrity/core_api/src/main/java/com/demonstration/table/coreapi/keys/MediatorKey.java
package com.demonstration.table.coreapi.keys;

import com.demonstration.table.coreapi.CoreMediator;

import java.lang.annotation.ElementType;
import java.lang.annotation.Target;

import dagger.MapKey;

@MapKey
@Target(ElementType.METHOD)
public @interface MediatorKey {
    Class<? extends CoreMediator> value();
}

File: modules/celebrity/core_api/src/main/java/com/demonstration/table/coreapi/keys/MediatorKey.kt
package com.demonstration.table.coreapi.keys

import com.demonstration.table.coreapi.CoreMediator
import dagger.MapKey
import kotlin.reflect.KClass

@MapKey
@Target(AnnotationTarget.FUNCTION)
annotation class MediatorKey(
    val values: KClass<out CoreMediator>
)